sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/NetworkInterfaceOperationsTests.java

```diff
@@ -509,6 +511,197 @@ public void canAssociateNatGateway() {
         Assertions.assertEquals(gateway2.id(), subnet2.natGatewayId());
     }
 
+    @Test
+    public void canCreateIpConfigurationsWithDeleteOption() {
```

Comment:
Can we combine all these tests into one? We can have more than one secondary Ip Configurations for a single NetworkInterface.

e.g.
```java
NetworkInterface nic = networkManager.networkInterfaces()
      .define(generateRandomResourceName("nic-", 15))
      .withRegion(Region.US_EAST)
      .withExistingResourceGroup(rgName)
      .withExistingPrimaryNetwork(vnet)
      .withSubnet(subnetName)
      .withPrimaryPrivateIPAddressDynamic()
      .withNewPrimaryPublicIPAddress()
      .withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
      .defineSecondaryIPConfiguration("secondary")
        .withExistingNetwork(vnet)
        .withSubnet(subnetName)
        .withPrivateIpAddressDynamic()
        .withNewPublicIpAddress()
        .withPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
        .attach()
      .defineSecondaryIPConfiguration("secondary2")
        .withExistingNetwork(vnet)
        .withSubnet(subnetName)
        .withPrivateIpAddressDynamic()
        .withNewPublicIpAddress()
        .withPublicIPAddressDeleteOptions(DeleteOptions.DELETE)
        .attach()
      .create();

// make assertions here

// test updateStage, updateDefinitionStage
nic.update()
   .withPrimaryPublicIPAddressDeleteOptions(DeleteOptions.DETACH)
   .updateIPConfiguration("secondary1")
     ...()
     .parent()
   .defineSecondaryIpConfiguration("secondaryN")
   .withPublicIPAddressDeleteOptions()
   .attach();

// make assertions here
```